src/incubator/TextField/types.ts

```diff
@@ -164,6 +164,10 @@ export type TextFieldProps = MarginModifiers &
   // FieldStateProps &
   ValidationMessageProps &
   Omit<CharCounterProps, 'maxLength' | 'testID'> & {
+    /**
+     * Pass id to the container
+     */
+    containerId?: string;
```

Comment:
See my other comment about the name :) 
Also, please put it after the `containerStyle` prop and also add it to the `textField.api.json` file (this file is for the docs site)